whois-commons/src/main/java/net/ripe/db/whois/common/apiKey/ApiKeyUtils.java

```diff
@@ -44,21 +50,22 @@ public static boolean validateScope(final OAuthSession oAuthSession, final List<
                     && maintainers.stream().anyMatch( maintainer -> scopeFormatter.getScopeKey().equalsIgnoreCase(maintainer.getKey().toString()));
     }
 
+
     private static Optional<String> getWhoisScope(OAuthSession oAuthSession) {
         final List<String> scopes = Arrays.asList(StringUtils.split(oAuthSession.getScope(), " "));
         return scopes.stream().filter(scope -> scope.startsWith("whois")).findFirst();
     }
 
-    public static boolean hasValidApiKey(final OAuthSession oAuthSession, final List<RpslObject> maintainers, final List<RpslAttribute> authAttributes) {
-        if(oAuthSession == null || oAuthSession.getUuid() == null) {
+    public static boolean hasValidApiKey(final OAuthSession oAuthSession, final List<RpslObject> maintainers, final List<RpslAttribute> authAttributes, final String environment) {
```

Comment:
If you are adding in the scope validation should be at scope level in method validateScope not here